ci.js

```diff
@@ -0,0 +1,181 @@
+
+        if (file.isDirectory()) {
+            await getFiles(`${currentPath}${file.name}/`, files);
+        } else if (file.name === "package.json") {
+            const packageJsonPath = path.join(
+                __dirname,
+                currentPath,
+                file.name
+            );
+            files.push(packageJsonPath);
+        }
+    }
+}
+
+async function doGetFiles(path) {
+    const files = [];
+    await getFiles(path, files);
+    return files;
+}
+
+async function getVersion(file) {
+    const content = await fs.readFile(file);
+    const json = JSON.parse(content);
+    return [json.name, json.version];
+}
+
+function processDep(obj, name, fn) {
+    if (!obj) {
+        return;
+    }
+
+    if (!obj[name]) {
+        return;
+    }
+
+    if (!/^workspace\:/.test(obj[name])) {
+        return;
+    }
+
+    const version = obj[name].replace("workspace:", "");
+    fn(obj, version);
+}
+
+async function getVersionsMap(allPackageJsons) {
+    const map = new Map();
+    for (let file of allPackageJsons) {
+        // console.log("Reading version from: ", file);
+        const [name, version] = await getVersion(file);
+        map.set(name, version);
+    }
+    return map;
+}
+
+function getVersionForPackageOrThrow(versionsMap, packageName) {
+    const version = versionsMap.get(packageName);
+    if (!version) {
+        console.log("Failed to get version for package: ", packageName);
+        process.exit(1);
+    }
+    return version;
+}
+
+async function checkConsistency(file, versionsMap) {
+    console.log("Checking: ", file);
+    const content = await fs.readFile(file);
+    const json = JSON.parse(content);
+    const version = getVersionForPackageOrThrow(versionsMap, json.name);
+
+    for (const [name, versionInDep] of versionsMap) {
+        const check = (x, version) => {
+            if (version.includes("*")) {
+                return;
+            }
+
+            if (versionInDep !== version) {
+                console.log(
+                    `Error, versions don't match: ${name}:${version} !== ${versionInDep}`,
+                    file
+                );
+                process.exit(1);
+            }
+        };
+        processDep(json.dependencies, name, check);
+        processDep(json.devDependencies, name, check);
+    }
+}
+
+async function bumpVersions(file, versionsMap) {
+    console.log("Updating: ", file);
+    let content = await fs.readFile(file);
+    const json = JSON.parse(content);
+
+    // bump dependencies
+    for (const [name, version] of versionsMap) {
+        const update = (x) => (x[name] = `workspace:${version}-${postfix}`);
+        processDep(json.dependencies, name, update);
+        processDep(json.devDependencies, name, update);
+    }
+
+    // also bump version in package itself
+    const version = getVersionForPackageOrThrow(versionsMap, json.name);
+    json.version = `${version}-${postfix}`;
+
+    content = JSON.stringify(json, undefined, 4) + "\n";
```

Comment:
no real need to use `let`. I think it's better to stick to immutability where possible to make it a tiny bit easier to follow in the future so my suggestion is to just have

const newContent = JSON.stringify(json, undefined, 4) + "\n";

so nobody ever questions why and where this variable is being mutated